File: Divisibility_&_Modular_Arithmetic/Q1. Ugly Number/Q1_UglyNumber.py
class Solution(object):
    def isUgly(self, n):
        """
        :type n: int
        :rtype: bool
        """
        if n <= 0:               # Step 1: Negative numbers and 0 are not ugly
            return False

        for p in [2, 3, 5]:      # Step 2: Check prime factors 2, 3, and 5
            while n % p == 0:    # Divide n by p as long as possible
                n //= p

        return n == 1            # Step 3: If n becomes 1, it’s an ugly number

File: Divisibility_&_Modular_Arithmetic/Q1. Ugly Number/Q1_UglyNumber.java
class Solution {
    public boolean isUgly(int n) {
        // Step 1: Negative numbers and 0 are not ugly
        if (n <= 0) return false;
        
        // Step 2: Divide n by 2, 3, and 5 as long as possible
        while (n % 2 == 0) n /= 2;  // Remove factor 2
        while (n % 3 == 0) n /= 3;  // Remove factor 3
        while (n % 5 == 0) n /= 5;  // Remove factor 5
        
        // Step 3: If n becomes 1, it has no other prime factors → ugly
        return n == 1;
    }
}
